addons/mail/models/mail_message.py

```diff
@@ -1016,13 +1016,19 @@ def _message_format(self, fnames):
     def message_fetch_failed(self):
         """Returns first 100 messages, sent by the current user, that have
         errors, in the format expected by the web client."""
-        messages = self.search([
-            ('has_error', '=', True),
-            ('author_id', '=', self.env.user.partner_id.id),
-            ('res_id', '!=', 0),
-            ('model', '!=', False),
-            ('message_type', '!=', 'user_notification')
-        ], limit=100)
+        author_id = self.env.user.partner_id.id
+        query = """
```

Comment:
You should call `self.flush` for the models/fields you are scanning in the raw SQL queries. If there are pending updates in the cache not yet flushed to the database, you could get inconsistent results.

Also, raw SQL queries bypass ir.rules so you may be missing some conditions in the WHERE clause and access restricted rows. You could use `self._search([])` to first build a Query object with the record rules applied (check odoo/odoo#143543)